gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/mfa_challenge.html

```diff
@@ -60,6 +60,8 @@
                     </div>
                     <input type="hidden" th:name="${_csrf.parameterName}" th:value="${_csrf.token}"/>
                     <input type="hidden" th:name="factorId" th:value="${factor.id}"/>
+                    <input th:if="${factor.factorType.type == 'RECOVERY CODE'}" type="hidden" th:name="recoveryCodefactorId" th:value="${factor.id}"/>
```

Comment:
Not a huge fan of this if ..., we have already the factorId one line above.

Reply:
Seems to not be used anywhere

Reply:
removed as it is not used